flagscale/serve/core/dag.py

```diff
@@ -0,0 +1,455 @@
+import os, importlib
+import sys
+import uvicorn
+import socket
+import subprocess
+import json
+import numpy as np
+import omegaconf
+import matplotlib.pyplot as plt
+
+from pathlib import Path
+from pydantic import create_model
+from typing import Callable, Any
+from fastapi import FastAPI, HTTPException, Request
+import ray
+from ray import workflow
+
+from flagscale.logger import logger
+
+
+class Builder:
```

Comment:
Builder for what? I believe the name should be more specific.

Reply:
done